controllers/akodeploymentconfig/cluster/cluster_controller_addon_secret.go

```diff
@@ -341,3 +350,28 @@ func getAKOPackageRefFromClusterBootstrap(log logr.Logger, cb *runv1alpha3.Clust
 	}
 	return -1, nil
 }
+
+func validateADCAndClusterIpFamily(cluster *clusterv1.Cluster, adc *akoov1alpha1.AKODeploymentConfig, isVIPProvider bool, log logr.Logger) error {
+	adcIpFamily := "V4"
+	if adc.Spec.ExtraConfigs.IpFamily != "" {
+		adcIpFamily = adc.Spec.ExtraConfigs.IpFamily
+	}
+	nodeIpFamily, err := utils.GetClusterIPFamily(cluster)
+	if err != nil {
+		log.Error(err, "can't get cluster ip family")
+		return err
+	}
+	if (adcIpFamily == "V4" && nodeIpFamily == "V6") || (adcIpFamily == "V6" && nodeIpFamily == "V4") {
+		errInfo := "We are not allowed to create single stack " + nodeIpFamily + " cluster when configure AKO as " + adcIpFamily + " ip family"
```

Comment:
Never use something like `we` , `I` in the error message. It could be something like:
```
AKO with IP family <> can not work together with cluster with IP family <>
```

Reply:
make sense. noted.